File: packages/root-detection/ios/Plugin/Classes/Results/IsEmulatorResult.swift
import Foundation
import Capacitor

@objc public class IsEmulatorResult: NSObject, Result {
    let emulator: Bool

    init(emulator: Bool) {
        self.emulator = emulator
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["emulator"] = emulator
        return result as AnyObject
    }
}

File: packages/root-detection/android/src/main/java/io/capawesome/capacitorjs/plugins/rootdetection/classes/results/IsEmulatorResult.java
package io.capawesome.capacitorjs.plugins.rootdetection.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.rootdetection.interfaces.Result;

public class IsEmulatorResult implements Result {

    private final boolean emulator;

    public IsEmulatorResult(boolean emulator) {
        this.emulator = emulator;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("emulator", emulator);
        return result;
    }
}
